library/core/src/iter/adapters/cloned.rs

```diff
@@ -134,3 +134,23 @@ where
     T: Clone,
 {
 }
+
+#[unstable(issue = "none", feature = "inplace_iteration")]
+unsafe impl<S, I> SourceIter for Cloned<I>
+where
+    Self: Iterator,
+    I: SourceIter<Source = S>,
+    S: Iterator,
+{
+    type Source = S;
+
+    #[inline]
+    unsafe fn as_inner(&mut self) -> &mut S {
+        // SAFETY: unsafe function forwarding to unsafe function with the same requirements
+        unsafe { SourceIter::as_inner(&mut self.it) }
+    }
+}
+
+// SAFETY: StepBy can only yield fewer items than its inner iterator
```

Comment:
How is `StepBy` related here?

Reply:
I noticed you marked this as resolved but either you messaged me somewhere else and I forgot or I missed the resolution. Can you remind me what the relation was?

Reply:
I resolved it by removing the comment which was a copy-paste error from a previous attempt to specialize `StepBy` with similar traits, but those specializations weren't tenable.